runtime/master/src/main/java/edu/snu/nemo/runtime/master/scheduler/BatchSingleJobScheduler.java

```diff
@@ -164,8 +170,8 @@ public void onTaskStateChanged(final String executorId, final String taskId,
         default:
           throw new UnknownExecutionStateException(new Exception("This TaskState is unknown: " + newState));
       }
-    } else if (attemptIdx < currentTaskAttemptIndex) {
-      // We can ignore this late arriving message.
+    } else if (taskAttemptIndex < currentTaskAttemptIndex) {
+      // Do not change state, as this notification is for a previous task attempt.
```

Comment:
Could you add an example case when this case can happens?
This means that a task notified its scheduling attempt which is 'previous' one, thus gets ignored.
When such case can occur?